valeriyapetrova/hometask-4/my_hash.rb

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+# looks like an hash
+class MyHash
+  include Enumerable
+  DEFAULT_MY_HASH_SIZE = 10
+  INIT_MY_HASH_COUNT = 0
+  CAPACITY = 0.7
+
+  def initialize
+    @size = DEFAULT_MY_HASH_SIZE
+    @count = INIT_MY_HASH_COUNT
+    @current_custom_hash = Array.new(DEFAULT_MY_HASH_SIZE) { [] }
+  end
+
+  def []=(key_input, value_input)
+    resize if (@count.to_f / @size) > CAPACITY
+    pair = @current_custom_hash[key_hash(key_input) % @size]&.find { |b| b[0] == key_input }
+    if pair
+      pair[1] = value_input
+    else
+      @count += 1
+      @current_custom_hash[key_hash(key_input) % @size] << [key_input, value_input]
+    end
+  end
+
+  def [](key_input)
+    pair = @current_custom_hash[key_hash(key_input) % @size]&.find { |b| b[0] == key_input }
+    pair ? pair[1] : nil
+  end
+
+  def delete(key_input)
+    pair = @current_custom_hash[key_hash(key_input) % @size]&.find { |b| b[0] == key_input }
```

Comment:
code here and in `def [](key_input)` and `def []=(key_input, value_input)`  are very similar, let's move out the common

Reply:
let's make `@current_custom_hash[key_hash(key_input) % @size]` expression more readable